src/charm.py

```diff
@@ -247,6 +250,12 @@ def _common_install_code(self) -> bool:
             logger.error("Failed to setup runner manager user")
             raise
 
+        try:
+            manager_service.install_package()
```

Comment:
What about upgrading? How is this handled?

Reply:
It is a local python package.
Installing it with code changes would update the installation.

The `_common_install_code` function is called on charm upgrade as well. So it should cover the charm upgrade event.

Reply:
So due to the `--ignore-installed` flag it would just install without caring about a previous installation?

Reply:
I have updated the `install_package` method to uninstall the packages prior to installing them.

Reply:
In the upgrade scenario, I guess the service needs to be stopped before "uninstall / install"